packages/x-tree-view/src/internals/plugins/useTreeViewSelection/useTreeViewSelection.types.ts

```diff
@@ -1,14 +1,49 @@
 import * as React from 'react';
-import type { DefaultizedProps, TreeViewItemRange, TreeViewPluginSignature } from '../../models';
+import type { DefaultizedProps, TreeViewPluginSignature } from '../../models';
 import { UseTreeViewItemsSignature } from '../useTreeViewItems';
 import { UseTreeViewExpansionSignature } from '../useTreeViewExpansion';
 
 export interface UseTreeViewSelectionInstance {
   isItemSelected: (itemId: string) => boolean;
-  selectItem: (event: React.SyntheticEvent, itemId: string, multiple?: boolean) => void;
-  selectRange: (event: React.SyntheticEvent, items: TreeViewItemRange, stacked?: boolean) => void;
-  rangeSelectToFirst: (event: React.KeyboardEvent, itemId: string) => void;
-  rangeSelectToLast: (event: React.KeyboardEvent, itemId: string) => void;
+  selectItem: (
+    event: React.SyntheticEvent,
+    itemId: string,
+    keepExistingSelection?: boolean,
+  ) => void;
+  /**
```

Comment:
I split the API into several methods instead of a single `selectRange`
That way each one can receive the parameters it needs AND we can apply some optimization (`selectAllNavigableItems` is a lot cheaper now, no need to try to reconcile the old and the new model, you just select everything).

Reply:
It makes a lot more sense like this! 👌 The previous API felt a little bit like mental gymnastics to me, this feels a lot more coherent 💯